### PR TITLE
Parse the updated model configuration after re-normalization by core

### PR DESCRIPTION
**Pull request by tanmayv25**
Also, abstracts some SetModelConfig logic into backend utilities.

### CODE REVIEW

File: src/backend_model.cc [hunk 1]
```diff
@@ -88,67 +83,115 @@ BackendModel::BackendModel(
...
   }
+
+  return nullptr;
 }
 
+TRITONSERVER_Error*
+BackendModel::SetModelConfig()
+{
+  triton::common::TritonJson::WriteBuffer json_buffer;
+  ModelConfig().Write(&json_buffer);
```
Comment (rmccorm4) [inline src/backend_model.cc:178]: tritonjson::Write returns TRITONSERVER_Error* for invalid json, so we should check it here with RETURN_IF_ERROR

**Review by rmccorm4 — CHANGES_REQUESTED** [review]
LGTM other than 1 comment

File: src/backend_model.cc [hunk 2]
```diff
@@ -88,67 +83,115 @@ BackendModel::BackendModel(
   THROW_IF_BACKEND_MODEL_ERROR(
       TRITONBACKEND_BackendMemoryManager(backend, &triton_memory_manager_));
 
+  THROW_IF_BACKEND_MODEL_ERROR(ParseModelConfig());
+}
+
+TRITONSERVER_Error*
+BackendModel::ParseModelConfig()
+{
```
Comment (nv-kmcgill53) [inline src/backend_model.cc:91]: From here to line 107 is a copy of lines 40-55 above. I think you forgot to delete the old code?

File: src/backend_model.cc [hunk 3]
```diff
@@ -88,67 +83,115 @@ BackendModel::BackendModel(
...
+
+  // Triton core can normalize the missing config settings
+  // in the above call. We must retrieve the updated model
+  // configration from the core.
+  RETURN_IF_ERROR(ParseModelConfig());
+
+  return nullptr;
+}
+
+
```
Comment (krishung5) [inline src/backend_model.cc:194]: Extra line?

**Review by krishung5 — COMMENTED** [review]
Only a minor comment, otherwise LGTM

**Review by rmccorm4 — APPROVED** [review]

**Pull request merged**